### PR TITLE
Add python wrappers for MappingQGeneric

### PR DESCRIPTION
**Pull request by agrayver**
This covers only MappingQGeneric, but I believe for pre-processing purposes this should suffice most of the needs (like running some GridTools functions, etc). Anyway, one has to start with something. 

@Rombur @masterleinad FYI

Part of #9015

### CODE REVIEW

**Comment by agrayver** [comment]
Additionally, this fixed some bugs in cell and tria accessors.

**Comment by bangerth** [comment]
For future reference, it's generally easier to review patches (and from your perspective to get them merged) if you create separate pull requests for separate issues -- here, one for the new functionality, and one for the bug fix.

As for the content, I'll leave that to those who actually understand the Python bindings :-)

File: contrib/python-bindings/source/cell_accessor_wrapper.cc [hunk 1]
```diff
@@ -167,7 +167,7 @@ namespace python
         static_cast<const CellAccessor<dim, spacedim> *>(cell_accessor);
       Point<spacedim>     barycenter = cell->barycenter();
       boost::python::list barycenter_list;
-      for (int i = 0; i < dim; ++i)
+      for (int i = 0; i < spacedim; ++i)
```
Comment (agrayver) [inline contrib/python-bindings/source/cell_accessor_wrapper.cc]: This is a bug fix.

File: contrib/python-bindings/source/tria_accessor_wrapper.cc [hunk 2]
```diff
@@ -34,7 +34,7 @@ namespace python
           tria_accessor);
       Point<spacedim>     barycenter = accessor->barycenter();
       boost::python::list barycenter_list;
-      for (int i = 0; i < dim; ++i)
+      for (int i = 0; i < spacedim; ++i)
```
Comment (agrayver) [inline contrib/python-bindings/source/tria_accessor_wrapper.cc]: This is a bug fix.

**Comment by agrayver** [comment]
@bangerth makes sense. For now, I have pointed out the fixes.

File: contrib/python-bindings/source/mapping_wrapper.cc [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+      Point<spacedim> p_real =
+        mapping->transform_unit_to_real_cell(cell, *point);
+
+      boost::python::list coord_list;
+      for (int i = 0; i < spacedim; ++i)
+        coord_list.append(p_real[i]);
+
+      return PointWrapper(coord_list);
+    }
+
```
Comment (Rombur) [inline contrib/python-bindings/source/mapping_wrapper.cc]: Add two more blank lines here please.

File: contrib/python-bindings/source/mapping_wrapper.cc [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+      Point<dim> p_real = mapping->transform_real_to_unit_cell(cell, *point);
+
+      boost::python::list coord_list;
+      for (int i = 0; i < dim; ++i)
+        coord_list.append(p_real[i]);
+
+      return PointWrapper(coord_list);
+    }
+  } // namespace internal
+
```
Comment (Rombur) [inline contrib/python-bindings/source/mapping_wrapper.cc]: Add blank lines. Same below.

File: contrib/python-bindings/source/mapping_wrapper.cc [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <cell_accessor_wrapper.h>
+#include <mapping_wrapper.h>
```
Comment (Rombur) [inline contrib/python-bindings/source/mapping_wrapper.cc]: I can't find this file. Did you forget to push it?

File: contrib/python-bindings/tests/mapping_wrapper.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+class TestMappingWrapper(unittest.TestCase):
+
+    def setUp(self):
+        self.triangulation = Triangulation('2D')
+        self.triangulation.generate_hyper_cube()
+        self.triangulation.refine_global(1)
+
+    def test_mapping(self):
+        mapping = MappingQGeneric(2,2,1)
```
Comment (Rombur) [inline contrib/python-bindings/tests/mapping_wrapper.py]: Can you add a test with `spacedim > dim`, just to make sure that we got all the `dim` and `spacedim` correct.

**Comment by agrayver** [comment]
@Rombur thanks, I addressed your comments.

**Comment by masterleinad** [comment]
You need to fix the indentation.

**Comment by agrayver** [comment]
@masterleinad done

**Comment by agrayver** [comment]
Can this be merged?

**Comment by agrayver** [comment]
ping

**Pull request merged**